lib/readline.js

```diff
@@ -182,6 +186,12 @@ function Interface(input, output, completer, terminal) {
     this._ttyWrite = _ttyWriteDumb.bind(this);
   }
 
+  function onerror(err) {
+    const debug = require('internal/util/debuglog').debuglog('stream');
+    debug(err);
+    throw new ERR_STREAM_PREMATURE_CLOSE();
```

Comment:
Can you give some information about why this is better than the way errors are currently handled? This adds an exception that is unlikely to be caught, and more importantly, information about the original error is lost here (except for the `debug()` call, but usually users don’t have debugging turned on for readline).

It would also be good to have a test for this error, to make sure it’s emitted correctly.

Reply:
Currently, there are no 'error' listeners for errors in the underlying buffer for the input stream in the readline module, as stated in the issue, and this adds that functionality. Regarding the original error info, I went with `ERR_STREAM_PREMATURE_CLOSE` but I can change that to `throw err` so as to propagate the original error back up.

> It would also be good to have a test for this error, to make sure it’s emitted correctly.

Sure! I'll push one right away!